### PR TITLE
feat: Add radio button page

### PR DESCRIPTION
**Pull request by williamernest**
resolves: #30

### CODE REVIEW

File: src/RadioButtonPage.js [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+const RadioButtonPage = () => {
+  return (
+      <div>
+        <HeaderBar />
+        <ComponentPage
+            hero={<RadioButtonHero/>}
+            title='Radio Button'
+            description='Buttons communicate an action a user can take. They are typically placed throughout your UI, in places like dialogs, forms, cards, and toolbars.'
+            designLink='https://material.io/guidelines/components/buttons.html'
```
Comment (moog16) [inline src/RadioButtonPage.js]: Update these links to radio

File: src/RadioButtonPage.js [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+            demos={<RadioButtonDemos/>}
+        />
+      </div>
+  );
+}
+
+class RadioButtonHero extends Component {
+  constructor(props) {
+    super(props);
+    this.ripples = [];
```
Comment (moog16) [inline src/RadioButtonPage.js]: moves these out of the constructor and delete constructor.

File: src/RadioButtonPage.js [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+    this.initRipple = buttonEl => this.ripples.push(new MDCRipple(buttonEl));
+  }
+
+  componentWillUnmount() {
+    this.ripples.forEach(ripple => ripple.destroy());
+  }
+
+  render() {
+    return (
+        <div>
```
Comment (moog16) [inline src/RadioButtonPage.js]: 2 spaces instead of 4 here

File: src/RadioButtonPage.js [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+          <div className='mdc-radio'>
+            <input className='mdc-radio__native-control' type='radio' id='hero-radio-1' name='radios' defaultChecked={true} />
```
Comment (moog16) [inline src/RadioButtonPage.js]: if it is a boolean you can omit the {true}

Should be: 

```
<input defaultChecked />
```

File: src/RadioButtonPage.js [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+          </div>
+        </div>
+    );
+  }
+}
+
+class RadioButtonDemos extends Component {
+  constructor(props) {
+    super(props);
+    this.setCount = 0;
```
Comment (moog16) [inline src/RadioButtonPage.js]: move setCount to the class, and delete constructor.

File: src/RadioButtonPage.js [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+  }
+
+  renderRadioButtonVariant(title) {
+    this.setCount++;
+    let radioCount = 0;
+    return (
+        <div className='demo-example'>
```
Comment (moog16) [inline src/RadioButtonPage.js]: tab indent

File: src/RadioButtonPage.js [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+          <h3 className='mdc-typography--subtitle2'>{title}</h3>
+          <div className='mdc-form-field'>
+            <div className='mdc-radio'>
+              <input className='mdc-radio__native-control' type='radio' id={'radio-' + radioCount} name={'radio-set-' + this.setCount} defaultChecked={true} />
```
Comment (moog16) [inline src/RadioButtonPage.js]: use interpolated strings

File: src/RadioButtonPage.js [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+              <div className='mdc-radio__background'>
+                <div className='mdc-radio__outer-circle'/>
+                <div className='mdc-radio__inner-circle'/>
+              </div>
+            </div>
+            <label id='radio-1-label' htmlFor='radio-1'>Radio 1</label>
+            <div className='mdc-radio'>
+              <input className='mdc-radio__native-control' type='radio' id={'radio-' + radioCount} name={'radio-set-' + this.setCount} />
```
Comment (moog16) [inline src/RadioButtonPage.js]: interpolated strings

File: src/RadioButtonPage.js [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+            </div>
+            <label id='radio-2-label' htmlFor='radio-2'>Radio 2</label>
+          </div>
+        </div>
+    );
+  }
+
+  render() {
+    return (
+        <div>
```
Comment (moog16) [inline src/RadioButtonPage.js]: tab indent

File: src/RadioButtonPage.js (hunk 1 above)
Comment (moog16) [reply src/RadioButtonPage.js]: https://github.com/material-components/material-components-web/tree/master/packages/mdc-radio

File: src/RadioButtonPage.js (hunk 7 above)
Comment (moog16) [reply src/RadioButtonPage.js]: actually pull them out into a variable like `id` and `name` to be used in 2 places

File: src/RadioButtonPage.js [hunk 10]
```diff
@@ -0,0 +1,101 @@
...
+          <h3 className='mdc-typography--subtitle2'>{title}</h3>
+          <div className='mdc-form-field'>
+            <div className='mdc-radio'>
+              <input className='mdc-radio__native-control' type='radio' id={'radio-' + radioCount} name={'radio-set-' + this.setCount} defaultChecked={true} />
+              <div className='mdc-radio__background'>
+                <div className='mdc-radio__outer-circle'/>
+                <div className='mdc-radio__inner-circle'/>
+              </div>
+            </div>
+            <label id='radio-1-label' htmlFor='radio-1'>Radio 1</label>
```
Comment (moog16) [inline src/RadioButtonPage.js]: On the appspot demo pages, if I click the label, it also toggles the radio button. Here that functionality is lost.

Comment (moog16) [reply src/RadioButtonPage.js]: @williamernest please fix. Github is hiding this comment.

File: src/RadioButtonPage.js [hunk 11]
```diff
@@ -0,0 +1,97 @@
...
+
+const RadioButtonPage = () => {
+  return (
+    <div>
+      <HeaderBar />
+      <ComponentPage
+        hero={<RadioButtonHero/>}
+        title='Radio Button'
+        description='Buttons communicate an action a user can take. They are typically placed throughout your UI, in places like dialogs, forms, cards, and toolbars.'
+        designLink='https://material.io/guidelines/components/buttons.html'
```
Comment (lynnjepsen) [inline src/RadioButtonPage.js]: still pointing to button

File: src/styles/RadioButtonPage.scss [hunk 12]
```diff
@@ -0,0 +1,2 @@
+@import "@material/radio/dist/mdc.radio";
+@import "@material/form-field/dist/mdc.form-field";
```
Comment (lynnjepsen) [inline src/styles/RadioButtonPage.scss]: add a little bit more space between the "Radio 1" and "Radio 2" so the ripples do not overlap

**Pull request merged**